### PR TITLE
Setup Cypress for end-to-end tests

### LINKED ISSUE #71: Setup Cypress for E2E testing
**Is your enhancement related to a problem? Please describe.**

See overall concept in the [OSBP guide](https://10up.github.io/Open-Source-Best-Practices/testing/#e2e-testing) and example setups on [Simple Podcasting](https://github.com/10up/simple-podcasting), [Restricted Site Access](https://github.com/10up/restricted-site-access), and https://github.com/10up/ElasticPress/pull/2446.

**Describe the solution you'd like**

  add base GitHub Action
  add .wp-env.json
  add base docker files
  update package.json / package lock files
  add cypress tests
  remove any WP Acceptance components

**Designs**

n/a

**Describe alternatives you've considered**

n/a

**Additional context**

https://github.com/10up/cypress-wp-setup/ && https://github.com/10up/cypress-wp-utils will be helpful in getting this spun up and some initial test commands to leverage.

### PR DESCRIPTION
**Pull request by cadic**
### Description of the Change

Added tests for:
- [x] Block Editor
- [x] Classic Editor

Closes #71 

### Checklist:

- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my change.
- [x] All new and existing tests passed.

### Credits

Props @

### CODE REVIEW

**Comment by jeffpaul** [comment]
@cadic want to open this up for review perhaps and then open a subsequent issue to document additional work to be done for e2e tests to keep this moving along?

**Comment by jeffpaul** [comment]
@cadic anything else needed in this PR before it can be moved out of draft for review/merge?

**Comment by cadic** [comment]
@jeffpaul this was staying on hold too long, waiting for classic editor functions to be added into our https://github.com/10up/cypress-wp-utils and now this PR is ready to proceed. It also needs some updates (upgrade Cypress from 9 to 10 or probably 11)

File: tests/cypress/support/commands.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+		cy.get(
+			'.components-button[aria-label="Options"], .components-button[aria-label="More options"]'
+		).click();
+	});
+	cy.get(".components-button").contains("Edit as HTML").click();
+});
+
+Cypress.Commands.add("insertInsecureBlock", (after) => {
+	cy.insertBlock("core/image").then((id) => {
+		cy.get(`#${id} .components-form-file-upload input[type=file]`).attachFile(
```
Comment (iamdharmesh) [inline tests/cypress/support/commands.js]: We may change `.attachFile()` to `.selectFile()` here and remove the `cypress-file-upload` dependency as `.selectFile()` is available from Cypress 9.3 

More info: https://docs.cypress.io/guides/references/migration-guide#Migrating-from-cypress-file-upload-to-selectFile

File: package.json [hunk 2]
```diff
@@ -22,31 +22,41 @@
...
+    "format": "npm run format-js",
+    "cypress:open": "cypress open --browser chrome --config-file tests/cypress/cypress.config.js",
+    "cypress:run": "cypress run --browser chrome --config-file tests/cypress/cypress.config.js",
+    "env": "wp-env",
+    "env:start": "wp-env start",
+    "env:stop": "wp-env stop",
+    "postenv:start": "./tests/bin/initialize.sh"
   },
   "dependencies": {
+    "cypress-file-upload": "^5.0.8",
```
Comment (iamdharmesh) [inline package.json]: We may remove this as we already have `.selectFile()` in Cypress and no longer needed this dependency. need to update `.attachFile()` to `.selectFile()`

More info: https://docs.cypress.io/guides/references/migration-guide#Migrating-from-cypress-file-upload-to-selectFile

**Review by iamdharmesh — COMMENTED** [review]
Thanks a lot for working on adding E2E tests. The code looks good to me. just added 2 minor nice-to-have comments. Please let me know what you think about that.

Thanks

File: tests/cypress/support/commands.js (hunk 1 above)
Comment (cadic) [reply tests/cypress/support/commands.js]: Thank you for this suggestion!

**Review by iamdharmesh — APPROVED** [review]
Thanks for changes @cadic, LGTM now 🚀

**Comment by jeffpaul** [comment]
@dkotter will leave up to you if you want to merge in ahead of the 1.0.3 release here

**Pull request merged**